presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/metadata/QueryFailureMetadata.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.spi.eventlistener.metadata;
+
+import com.facebook.presto.spi.ErrorCode;
+
+import java.util.Optional;
+
+public class QueryFailureMetadata
+{
+    private final Optional<ErrorCode> errorCode;
```

Comment:
Instead of making all fields optional, why not make the failureInfo entry in the query event optional?